### PR TITLE
lint: fix multi-part imports

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [X] Have you successfully run `make lint`?
- [X] Have you successfully run `pytest tests/unit`?

-----

Update from [`pyright 1.1.299`](https://github.com/microsoft/pyright/releases/tag/1.1.299):
> Fixed a recent regression that resulted in multi-part imports being evaluated as unknown if a namespace package was used for an intermediate part of the module path and useLibraryCodeForTypes was set to false.

In other words, pyright now complains if you do this:
```python
import xdg.BaseDirectory
CONFIG_PATH = Path(xdg.BaseDirectory.xdg_config_home) / "snapcraft/snapcraft.cfg"
```

### CODE REVIEW

**Review by tigarmo — APPROVED** [review]
thanks for the quick fix!
